s2core/src/main/scala/com/daumkakao/s2graph/core/Management.scala

```diff
@@ -59,39 +62,19 @@ object Management extends JSONParser {
    * copy label: only used by bulk load job. not sure if we need to parameterize hbase cluster.
    */
   def copyLabel(oldLabelName: String, newLabelName: String, hTableName: Option[String]) = {
-    Label.findByName(oldLabelName) match {
-      case Some(old) =>
-        Label.findByName(newLabelName) match {
-          case None =>
-            val idxSeqs = (LabelIndex.findByLabelIdAll(old.id.get).flatMap { idx =>
-              idx.metaSeqs
-            }).toSet
-            val (indexProps, metaProps) = old.metaPropsInvMap.partition { case (name, meta) => idxSeqs.contains(meta.seq) }
-
-            val idxPropsSeq = for {
-              (name, meta) <- indexProps
-            } yield {
-                (name, meta.defaultValue, meta.dataType)
-              }
-            val metaPropsSeq = for {
-              (name, meta) <- metaProps
-            } yield {
-                (name, meta.defaultValue, meta.dataType)
-              }
-
-            createLabel(newLabelName, old.srcService.serviceName, old.srcColumnName, old.srcColumnType,
-              old.tgtService.serviceName, old.tgtColumnName, old.tgtColumnType,
-              old.isDirected, old.serviceName,
-              idxPropsSeq.toSeq,
-              metaPropsSeq.toSeq,
-              old.consistencyLevel, hTableName, old.hTableTTL, old.schemaVersion, old.isAsync)
-
-          case Some(_) =>
-//            throw new LabelAlreadyExistException(s"New label $newLabelName already exists.")
-        }
-      case None => throw new LabelNotExistException(s"Original label $oldLabelName does not exist.")
-    }
+    val old = Label.findByName(oldLabelName).getOrElse(throw new LabelAlreadyExistException(s"Old label $oldLabelName already exists."))
+    if (Label.findByName(newLabelName).isDefined) throw new LabelAlreadyExistException(s"New label $newLabelName already exists.")
+
+    val allProps = old.metas.map { labelMeta => Prop(labelMeta.name, labelMeta.defaultValue, labelMeta.dataType) }
```

Comment:
.metas call LabelMeta.findAllByLabelId which will return only labelMetas in DB(without reserved labelMetas) so seems solve previous problem.